backend/controllers/users.js

```diff
@@ -179,69 +182,104 @@ usersRouter.post(
       });
     }
 
-  //Check user is exist or not. If not, return with error message.
-  // Update user register API
-  const registerUser = async (request, response) => {
+ // Check user is exist or not. If not, return with an error message.
+// Update user register API
+const registerUser = async (request, response) => {
   try {
-    // Check if the user already exists
     const userInfo = request.body;
-    const userExists = await User.findOne({ $or: [{ email: userInfo.email }, { username: userInfo.username }] });
 
+    // Check if the user already exists
+    const userExists = await User.findOne({
+      $or: [{ email: userInfo.email }, { username: userInfo.username }],
+    });
     if (userExists) {
       return response.status(400).json({
-        status: "Fail",
-        error: "User already exists",
+        status: 'Fail',
```

Comment:
Here you should check UserApp to confirm user is already connected to app or not